core/sdk-core/src/main/java/software/amazon/awssdk/core/SelectedAuthScheme.java

```diff
@@ -30,20 +30,20 @@
 @SdkProtectedApi
 public final class SelectedAuthScheme<T extends Identity> {
 
-    private IdentityProvider<T> identityProvider;
+    private CompletableFuture<? extends T> identity;
     private HttpSigner<T> signer;
     private AuthSchemeOption authSchemeOption;
 
-    public SelectedAuthScheme(IdentityProvider<T> identityProvider,
+    public SelectedAuthScheme(CompletableFuture<? extends T> identity,
                               HttpSigner<T> signer,
                               AuthSchemeOption authSchemeOption) {
-        this.identityProvider = Validate.paramNotNull(identityProvider, "identityProvider");
-        this.signer = Validate.paramNotNull(signer, "signer");
```

Comment:
removing the nonNull check (for smithy.api#noAuth use case) was initially separated out in https://github.com/aws/aws-sdk-java-v2/pull/4180, but folded it into this PR now.